dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1141,6 +1158,52 @@ CDefaultContextMenu::InvokeRegVerb(
 
     PStaticShellEntry pEntry = &m_StaticEntries.GetAt(it);
 
+    HKEY hVerbKey;
+    WCHAR VerbKey[sizeof("shell\\") + MAX_VERB];
+    hr = StringCbPrintfW(VerbKey, sizeof(VerbKey), L"shell\\%s", pEntry->Verb.GetString());
+    if (SUCCEEDED(hr) && m_pDataObj &&
+        RegOpenKeyExW(pEntry->hkClass, VerbKey, 0, KEY_READ, &hVerbKey) == ERROR_SUCCESS)
+    {
+        CLSID clsid;
+        BOOL Done = FALSE;
+
+        DWORD KeyState = 0;
+        if (lpcmi->fMask & CMIC_MASK_SHIFT_DOWN)
+            KeyState |= MK_SHIFT;
+        if (lpcmi->fMask & CMIC_MASK_CONTROL_DOWN)
+            KeyState |= MK_CONTROL;
+
+        POINTL *pPtl = NULL;
+        C_ASSERT(sizeof(POINT) == sizeof(POINTL));
+        if (lpcmi->fMask & CMIC_MASK_PTINVOKE)
+            pPtl = (POINTL*)&lpcmi->ptInvoke;
+
+        if (!Done)
```

Comment:
Not sure whether you intend to have some extra code before that block that will change the `Done` variable...
... because currently otherwise, this condition is always TRUE here.

Reply:
Yes, believe it or not, there is also `IExplorerCommand`! I have not investigated that one yet so I don't know where it goes.